### ISSUE TITLE
Full screen image appears on wrong monitor

### ISSUE THREAD

**Issue by UgnilJoZ** [Nheko-Reborn/nheko#1016]
### Describe the bug

When using nheko on a secondary screen, clicking on an image displays the image in fullscreen on the primary monitor.

### To Reproduce

1. Enter a Gnome Wayland session (Nvidia on my side) with a two-monitor setup
2. Put nheko on the secondary monitor
3. Select a chat
4. Click on an image to display it full-screen

### What happened?

The image is displayed in full screen on the other (primary) monitor.

(Subsequent clicks on the image open more and more image viewers on top of the first one.)

Additionally, sometimes on the first try after starting nheko, the image message is only selected instead of show in full screen.

### Expected behavior

I would have expected the image to appear as overlay on top of the chat window.

### Screenshots

![Primary monitor on the bottom, secondary on top](https://user-images.githubusercontent.com/7910828/160104906-22a4c1e6-683a-4fb5-8e1c-98725dd7f4c1.jpg)

### Version

0.9.3-9bac429

### Operating system

Linux

### Installation method

Flathub

### Did you use profiles?

- [ ] Profiles used?

### Relevant log output

```shell
No logs are generated during the event of showing images.
```

**Comment by ceandreasen** [comment]
I also am encountering this same issue on Pop OS 22.04 using X11, so it isn't just a wayland issue. Nheko opens on my primary monitor, but regardless of where I put the nheko window, the image viewer always pops up over my secondary display (and clicking the image again in confusion opens more image previews) 

I personally would think that further image previews shouldn't open if you keep clicking, unless there's a use case for that I'm missing, but that's obviously a lower priority/basically a non-issue if the popup was always covering the main UI.

**Comment by MayeulC** [comment]
I also have the same issue on sway/wayland. The fullscreen image viewer seems to favor my rightmost monitor currently, which is a tiny Steam Deck screen...

Also, it's weird that the window is a fullscreened borderless pop-up. I would have imagined a normal window, or a fullscreen window. Preferably with server-side decorations still (decorations are hidden when full-screen anyway).

<details><summary>

`swaymsg -t get_tree` output with nheko preview window

</summary>

```json
"floating_nodes": [
            {
              "id": 133,
              "type": "floating_con",
              "orientation": "none",
              "percent": 1.1010752688172043,
              "urgent": false,
              "marks": [],
              "focused": false,
              "layout": "none",
              "border": "csd",
              "current_border_width": 2,
              "rect": {
                "x": 2688,
                "y": 0,
                "width": 1280,
                "height": 800
              },
              "deco_rect": {
                "x": 0,
                "y": 0,
                "width": 0,
                "height": 0
              },
              "window_rect": {
                "x": 0,
                "y": 0,
                "width": 1280,
                "height": 800
              },
              "geometry": {
                "x": 0,
                "y": 0,
                "width": 500,
                "height": 500
              },
              "name": "nheko",
              "window": null,
              "nodes": [],
              "floating_nodes": [],
              "focus": [],
              "fullscreen_mode": 1,
              "sticky": false,
              "pid": 32405,
              "app_id": "io.github.NhekoReborn.Nheko",
              "visible": true,
              "max_render_time": 0,
              "shell": "xdg_shell",
              "inhibit_idle": false,
              "idle_inhibitors": {
                "user": "none",
                "application": "none"
              }
            }
          ],
```

</details>

**Comment by TomTheDragon** [comment]
The bug is still persistent on X11 and wayland.
Any updates on this?

**Comment by deepbluev7** [comment]
> The bug is still persistent on X11 and wayland. Any updates on this?

Sure: Nobody fixed it in Qt yet.

**Comment by ceandreasen** [comment]
> Sure: Nobody fixed it in Qt yet.

Thanks! Is there an upstream bug report we can track or work on?

**Comment by UgnilJoZ** [comment]
> Nheko uses a separate window to display images.

If it is not possible to control where it is opened, then may there be a better technical approach to the image preview than opening an additional window? Something like an overlay component within the existing window? 

> It should open on the same monitor, but we have no control over that on wayland.

There seems to be some kind of influence an application can exert.

If I click on an image in [Telegram desktop](https://github.com/telegramdesktop/tdesktop), it opens a new window on top the existing one and on the same monitor. (Unless the window spans over multiple screens, then it seems to be a little more random).

**Comment by forkiesassds** [comment]
How is this an upstream QT issue? Is there any issue on their issue tracker? I'm using KDE Plasma 6 in Wayland and I haven't encountered any problems where windows open on a different monitor than the one where my cursor is at other than with some X11 programs. But with this application I do encounter this issue. Originally I thought this was because for SOME REASON it was using X11, turns out not. Even making it forcefully use Wayland causes this issue.

**Comment by deepbluev7** [comment]
Actually, I can't find the issue anymore, that I thought was there. I thought there was one like https://bugreports.qt.io/browse/QTBUG-30724, but for Linux, but I can't find it anymore.

But for anyone interested in it, the code is here: https://github.com/Nheko-Reborn/nheko/blob/0c222689f47991bcc47fd68566deb82592e370e3/resources/qml/Root.qml#L254

Possibly the parent needs to be set differently to show on the same screen or something? I always assumed "showFullScreen" would just open on the same monitor. Possibly the transientParent needs to be set on the Window. In theory that property is set automatically though, but maybe that doesn't work if there is no Window above it in qml (and the window is instead the QQuickWindow created in C++).

**Comment by forkiesassds** [comment]
Wouldn't it be better if the image was just shown within the same window? Something like how Element and Discord does it. Doing so would probably eliminate a whole dozen of weird issues like these.

**Comment by deepbluev7** [comment]
Not for me, I usually put the image manually on a different monitor, so that it doesn't block the chat and I can discuss it.

**Comment by forkiesassds** [comment]
Then it's a feature, not a bug. :trollface:

**Comment by Aq32** [comment]
Can confirm that this also affects Windows and nheko version 0.12.1-dc910109
If you open an image with the chat in a secondary monitor the image viewer opens up on the primary monitor. Maybe there should be an option to choose whether to open the image in the active chat window monitor or some other monitor (which might be complicated by having more than 2 monitors).
